presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1348,11 +1356,27 @@ else if (expression instanceof LongLiteral) {
                     }
 
                     // otherwise, just use the expression as is
+                    boolean orderByOnExpression = false;
                     if (orderByExpression == null) {
                         orderByExpression = expression;
+                        orderByOnExpression = true;
                     }
 
                     ExpressionAnalysis expressionAnalysis = analyzeExpression(orderByExpression, sourceScope);
+                    if (orderByOnExpression) {
+                        // Try to resolve within orderByScope. If it is possible, then we warn - standard behavior differs from legacy one.
+                        for (Expression columnExpression : expressionAnalysis.getColumnReferences()) {
+                            Optional<ResolvedField> resolvedField = orderByScope.tryResolveField(columnExpression);
```

Comment:
`expressionAnalysis.getColumnReferences()` are any column references within an expression, take query for example
```
SELECT 1 as a FROM ... GROUP BY a ORDER BY (SELECT a FROM (VALUES 1) t(a))
```
order by expression:
```
(SELECT a FROM (VALUES 1) t(a))
```
will have column reference `a`. However, `a` is a correct reference to `(VALUES 1) t(a))` and not to output projection (e.g: `orderByScope.tryResolveField(columnExpression)` would be present). For proper verification, you should base the code on https://github.com/prestodb/presto/pull/7354 since it stores reference bindings to relation in `Analysis`.

The problem here is that you take arbitrary column reference out of context and apply order by scope, which might not be applicable for the particular reference